agent/engine/task_manager.go

```diff
@@ -542,3 +549,31 @@ func (mtask *managedTask) discardPendingMessages() {
 		}
 	}
 }
+
+var _stoppedSentWaitInterval = stoppedSentWaitInterval
+var _maxStoppedWaitTimes = int(maxStoppedWaitTimes)
+
+// waitForStopReported will wait for the task to be reported stopped and return true, or will time-out and return false.
+// Messages on the mtask.dockerMessages and mtask.acsMessages channels will be handled while this function is waiting.
+func (mtask *managedTask) waitForStopReported() bool {
+	stoppedSentBool := make(chan bool)
+	taskStopped := false
+	go func() {
+		for i := 0; i < _maxStoppedWaitTimes; i++ {
+			// ensure that we block until api.TaskStopped is actually sent
+			sentStatus := mtask.GetSentStatus()
+			if sentStatus >= api.TaskStopped {
+				taskStopped = true
+				break
+			}
+			seelog.Warnf("Blocking cleanup for task %v until the task has been reported stopped. SentStatus: %v (%d/%d)", mtask, sentStatus, i+1, _maxStoppedWaitTimes)
+			mtask._time.Sleep(_stoppedSentWaitInterval)
```

Comment:
* Why are you using `_time` as a member of mtask instead of just using `time.Sleep` directly?
* Why not use exponential backoff with the same 72 hour cap?